tests/components/ct-solid/playwright/index.tsx

```diff
@@ -1,12 +1,14 @@
 import '../src/assets/index.css';
 import { beforeMount, afterMount } from '@playwright/experimental-ct-solid/hooks';
+import { Router } from "@solidjs/router";
 
 export type HooksConfig = {
   route: string;
 }
 
-beforeMount<HooksConfig>(async ({ hooksConfig }) => {
+beforeMount<HooksConfig>(async ({ hooksConfig, App }) => {
   console.log(`Before mount: ${JSON.stringify(hooksConfig)}`);
+  return <Router><App /></Router>;
```

Comment:
@pavelfeldman Thinking out loud: Maybe it's useful that the plugins/wrappers for the JSX frameworks can be registered globally in the beforeEach hook, as this is also possible with Vue: `app.use(router)`. This ensures less code duplication and the tests themselves are less framework specific.

Reply:
Exposing wrapping facility certainly works for me. We can even make it explicit and introduce a third hook for it, but return value from the beforeMount is a good-enough start.

Reply:
Cool! Will create a follow up PR for the others

Reply:
Cool, i will develop it a bit further. I think a conscious choice has to be made how to deal with

Explored other possibilities:

## 1. `beforeMount` method 

### Solid
```tsx
import { beforeMount } from '@playwright/experimental-ct-solid/hooks';

beforeMount(async ({ App }) => {
  if (hooksConfig?.routing) return <Router><App /></Router>;
});
```

```tsx
test('works', async ({ mount }) => {
  await mount(<App />, { hooksConfig: { routing: true } });
});
```

### Vue

```tsx
import { beforeMount } from '@playwright/experimental-ct-solid/hooks';

beforeMount(async ({ app }) => {
  if (hooksConfig?.routing) app.use(router);
});
```

```tsx
test('works', async ({ mount }) => {
  await mount(<App />, { hooksConfig: { routing: true } });
});
```

---

## 2. `base.extend` method

### Solid
```tsx
import { test: base } from '@playwright/experimental-ct-solid';

const test = base.extend({
  mount({ mount }, use) => use((App) => mount(<Router><App /></Router>));
});
```

```tsx
test('works', async ({ mount }) => {
  await mount(<App />});
});
```


---


## 3. `beforeEach` method
### Solid
```tsx
import { test } from '@playwright/experimental-ct-solid';

test.beforeEach(async ({ App }) => <Router><App /></Router>);
```

```tsx
test('works', async ({ mount }) => {
  await mount(<Component />});
});
```

---


## 4. `factory` method

### Solid

```tsx
import { mount } from '@playwright/experimental-ct-solid';

function mountWithRouter(overrides = {}) {
  return mount(<Router><Component /></Router>)
}
```

```tsx
test('works', async () => {
  await mountWithRouter();
});

```

Reply:
Cool, will develop it a bit further! Explored some other possibilities as well. Takes a while to explain but _(without looking at technical limitations)_ i think the `beforeMount` hook is the best option of all.

## 1. `beforeMount` method 

```tsx
import { beforeMount } from '@playwright/experimental-ct-solid/hooks';

beforeMount(async ({ App }) => {
  if (hooksConfig?.routing) return <Router><App /></Router>;
});
```

```tsx
test('works', async ({ mount }) => {
  await mount(<App />, { hooksConfig: { routing: true } });
});
```

## 2. `base.extend` method

```tsx
import { test: base } from '@playwright/experimental-ct-solid';

const test = base.extend({
  mount({ mount }, use) => use((App) => mount(<Router><App /></Router>));
});
```

## 3. `beforeEach` method

```tsx
import { test } from '@playwright/experimental-ct-solid';

test.beforeEach(async ({ mount }) => mount(<Router><App /></Router>));
```

```tsx
test('works', async ({ mount }) => {
  await mount(<Component />});
});
```

## 4. `factory` method

```tsx
import { mount } from '@playwright/experimental-ct-solid';

function mountWithRouter(overrides = {}) {
  return mount(<Router><Component /></Router>)
}
```

```tsx
test('works', async () => {
  await mountWithRouter();
});
```

Reply:
I'm with you, beforeMount is the most straightforward and powerful given that it runs in the browser.